client/purplecaffeine/core.py

```diff
@@ -160,23 +168,52 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 class BaseBackend:
     """Base backend class."""
 
-    def save(self, name: str, trial: Trial):
+    def save(self, trial: Trial):
         """Saves given trial.
 
         Args:
-            name: name of the trial
             trial: encode trial to save
         """
-        raise NotImplementedError
+        requests.post(
```

Comment:
Wait :) Why are you changing base backend? 
Base backend should be a base class all other backends are inherited from. 

```python
class ApiBackend(BaseBackend):
```

Reply:
Oooh I misunderstood that, so you mean BaseBackend should never be implemented ?